pages/Messages/Validation.md

```diff
@@ -0,0 +1,173 @@
+
+## Specification Status
+
+| Version | Status |
+---------- | ---------
+| 0.1     | Tentative |
+
+## Purpose
+1. Describe the process of validating messages for inclusion a batch
+
+## Assumptions
+* All assumptions from [DSNP Messages](/Messages/Overview)
+* All assumptions from [DSNP Signatures](/Messages/Signatures)
+
+## General Message Validation
+
+In general, all implementations should validate all announcements for correctness against the [DSNP message schema](/Messages/Overview) and authenticity using [their provided signatures](/Messages/Signatures).
+Additionally, all client implementations should validate content for correctness against the [Activity Pub schema](https://www.w3.org/TR/activitypub/) and authenticity using the provided content hash.
+However, archivist implementations may choose to skip content validation steps in the interest of performance.
+
+### Announcement Correctness
+
+Validation of announcement correctness will vary depending on which fields are present on a particular DSNP message type.
+In general, correctness validation will consist of confirming that all necessary fields are present and that values in each field use a format appropriate for the field type.
+
+#### Encrypted Messages
+
+In the case of encrypted messages, if the validating user does not possess the necessary keys to decrypt a message, no validation is possible and the message should be treated as valid, i.e. included in a batch, unless proven otherwise.
+Clients and end-users who also do not possess the necessary keys should ignore these messages.
+
+Optionally, archivist implementations may choose to invalidate encrypted messages for which the content is obviously incorrect, i.e. the encrypted byte array isn't long enough to be a valid message, however no standard is defined for this behavior and caution should be used to avoid incorrectly invalidating messages.
+
+If the validating user does possess the necessary keys, the message should decrypted and validated as if it were a non-encrypted message.
+
+#### Non-Encrypted Messages
+
+Validating correctness for a non-encrypted message will consist of verifying that each field in the message is defined and meets the formatting rules listed in the subsections below for the given field type.
+
+##### Addresses
+
+1. Address fields must be exactly 20 bytes.
+
+##### Emoji
+
+1. Emoji fields must not be empty.
+1. Emoji fields must be less than 16 bytes.
+1. Emoji fields must only consist of [Unicode points](https://unicode.org/standard/standard.html) from `U+2000` to `U+2BFF`, from `U+E000` to `U+FFFF`, or from `U+1F000` to `U+10FFFF`.
+
+For example, all of the following should be considered valid emoji:
+
+```
+"😀", "🤌🏼", "👩🏻‍🎤", "🧑🏿‍🏫", "🏳️‍🌈", "🏳️‍⚧️", "⚛︎", "🃑", "♻︎"
+```
+
+Additionally, none of the following should be considered valid:
+
+```
+"F", ":custom-emoji:", "<custom-emoji>", "ᚱ", "ᘐ", "״"
+```
```

Comment:
I thought we wanted to allow for arbitrary length emoji to allow for merged emoji?

Reply:
Hmm... I think that works. I do wonder if we want to move some of this into the announcement type spec. We might have some reorg to do once some of the dust settles.